src/linux.rs

```diff
@@ -0,0 +1,85 @@
+use detect_desktop_environment::DesktopEnvironment;
+use std::path::Path;
+
+use anyhow::{Context, Result};
+use zbus::blocking::Connection;
+use zvariant::Value;
+
+use crate::Mode;
+
+fn get_freedesktop_color_scheme() -> Result<Option<Mode>> {
+    let conn = Connection::session()?;
+    let reply = conn.call_method(
+        Some("org.freedesktop.portal.Desktop"),
+        "/org/freedesktop/portal/desktop",
+        Some("org.freedesktop.portal.Settings"),
+        "Read",
+        &("org.freedesktop.appearance", "color-scheme"),
+    );
+    if let Ok(reply) = &reply {
+        let theme = reply.body::<Value>()?;
+        let theme = theme
+            .downcast::<u32>()
+            .with_context(|| "Failed to parse value")?;
+        match theme {
+            1 => Ok(Some(Mode::Dark)),
+            _ => Ok(Some(Mode::Light)),
+        }
```

Comment:
```suggestion
        match theme {
            1 => Ok(Some(Mode::Dark)),
            2 => Ok(Some(Mode::Light)),
            _ => Ok(None),
        }
```

I tested this in both KDE and GNOME on Fedora 35 and it always returned Light regardless of the theme. I discussed with some KDE developers and evidently this new dbus API has been implemented in both KDE and GNOME, but is not in stable releases of either yet. I used

```
qdbus org.freedesktop.portal.Desktop /org/freedesktop/portal/desktop org.freedesktop.portal.Settings.Read "org.freedesktop.appearance" "color-scheme"
```

to confirm there was not an issue with this code querying the dbus API; it was the dbus API returning 0. So evidently the return value is 0 when this is not implemented. In this case, I think this crate should fall back to the legacy methods, which returning `None` from this function will do.

Reply:
Excellent observation! I also tried using `qdbus` and got an error because the interface/key wasn't present in my system, for this reason I decided to return `None` when I got an error in the call. I didn't know 0 was returned when it is not implemented.

Reply:
Well, now it's prepared for both cases. 👍🏼